### DISCUSSION TITLE
Random cropOrpad - image and segmentation

### DISCUSSION THREAD

**Discussion by caumente** [TorchIO-project/torchio#569] [Q&A]
Hey everyone,

First of all, thanks for developing TorchIO. 

Currently I am working on a project related to multi-modal medical imaging segmentation on 3D. I load the images and their segmentation and so that I get two numpy.ndarrays with shapes:

image = ndarray(4, 155, 240, 240)
mask = ndarray(155, 240, 240)

In order to do data augmentation I would like to crop randomly the image and the mask to (128, 128, 128) spatial dimensions. Does cropOrpad do it randomly or centering? I suppose that if the crop is random, I couldn't crop them as follow

```python
import torchio as tio
transforms = tio.CropOrPad((128, 128, 128))
image, mask = transforms(image), transforms(mask)
```

I also wonder if it would be better do this kind of transformation into the dataloader or while training step on fly..

Thanks!

**Comment by fepegar** [comment]
Hi, @caumente. You can use a sampler and the queue. Please take a look at the docs for [Patch-based pipelines](https://torchio.readthedocs.io/data/patch_based.html) and let us know if that would help.
